### ISSUE TITLE
Content Request: Add video on VS Code REST Client

### ISSUE THREAD

**Issue by marcduiker** [marcduiker/azure-functions-university#11]
## 📝 Lesson Description

_Provide a clear and concise description of the content you'd like to have included._

This is not a full lesson on an Azure Functions topic but a short video on how to use the REST Client in VSCode when developing Azure Functions.

## Are you authoring the lesson? 

✍ _Indicate if you want to (help) write the lesson._

YES

👩‍🏫 _If you are authoring then indicate if you want to join our live stream to help teach this lesson._ 

YES

---

Thanks! 😃 That's all we need to know for now. If you're interested in writing the lesson yourself please read the __Process__ and __Lesson Structure__ parts below so you're well prepared! 💪

---

### Process

If you are contributing to create a lesson, please take into account the length and the tone of the existing lessons. We want to have a uniform experience across all of our lessons.

Once we've discussed this issue and agree to include the lesson you can submit a PR with the lesson and the source code. It will probably take a few iterations to get everything right, don't feel discouraged by this process ♥.

### Lesson Structure

Please keep in mind the following structure of a lesson:

- Give the lesson a short but descriptive title.
- Describe the goal of the lesson.
- Break up the lesson in small exercises.
- Each exercise has a clear sub goal and steps how to achieve that sub goal.
- Use code samples and plenty of 📝, 🔎 and ❔ call-outs.
- Add a new project to the `src` folder with a completely worked out example.

**Comment by rishabkumar7** [comment]
@marcduiker Just wondering if this lesson is just a demo for the [VSCode extension](https://marketplace.visualstudio.com/items?itemName=humao.rest-client) for REST Client and how you can utilize it to make requests to Azure Functions?

**Comment by madebygps** [comment]
@rishabkumar7 Yeah, I'm pretty sure that's correct.

**Comment by marcduiker** [comment]
@rishabkumar7 yes correct! These `tooling` videos are meant to demonstrate the tool specifically in combination with Azure Functions. I see them as secondary content though. I find the lessons on the triggers & bindings (in all languages) a bit more important.

**Comment by rishabkumar7** [comment]
@marcduiker For the lesson, should it be a video, and where does the code reside for the VSCode Rest Client demo?

**Comment by marcduiker** [comment]
Hi @rishabkumar7, there is no specific content for this lesson yet. I suggest to add:

- a `lesson\tooling\vscode-restclient` folder with a markdown file with written instructions how the REST client can be used (see further down for the list with topics).
- a `tst\tooling\vscode-restclient` folder with one or more `.http` files that contain the various REST code snippets to execute
- The video (that we record together) will then go over the markdown lesson and demonstrate it all live.

This repo can be used to run a Function App locally and make calls to it: https://github.com/marcduiker/functionapp-deployment. If you feel that a separate solution is better suited to demonstrate the REST client than a new FunctionApp can be added to `src/tooling/vscode-restclient`.

I suggest the following should be present in the lesson:
1. Why the REST Client is useful when working with HttpTriggered functions.
2. How to install the REST Client
4. How to do GET requests (incl query string)
5. How to do POST requests with a JSON body
6. Using file variables
7. Using environment variables
8. Using request variables
9. Using system variables (guid, randomInt, datetime)

**Comment by marcduiker** [comment]
@rishabkumar7 still interested in this one and would you be able to do this in Jan? If so, I'm happy to assign this to you. 

If not, let me know and I'll pick this one up myself.

**Comment by rishabkumar7** [comment]
@marcduiker yeah, I am interested in doing this one 👍🏽
